### PR TITLE
Replace `sqlite3` by `@cap-js/sqlite`

### PR DESCRIPTION
**Pull request by chgeo**

### CODE REVIEW

File: test/hierarchical-data.test.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ describe('cap/samples - Hierarchical Data', ()=>{
 	const {expect} = cds.test
 
 	before ('bootstrap sqlite in-memory db...', async()=>{
-		await cds.deploy (csn) .to ('sqlite::memory:') // REVISIT: cds.compile.to.sql should accept cds.compiled.for.nodejs models
+		// REVISIT: cds.compile.to.sql should accept cds.compiled.for.nodejs models
+		// REVISIT: how to better configure new sqlite
+		await cds.deploy (csn) .to ({impl: '@cap-js/sqlite'}) // 'sqlite::memory:'
```
Comment (chgeo) [inline test/hierarchical-data.test.js:20]: @danjoa @johannes-vogel this is needed today to make `cds.deploy` use the new sqlite service
A package.json with a dependency next to the test doesn't help, and is maybe not the best way either.
So: how shall such a test indicate it wants to use new sqlite.

File: test/messaging.test.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ describe('cap/samples - Messaging', ()=>{
     })
 
     it ('should bootstrap sqlite in-memory db', async()=>{
-        const db = await cds.deploy (_model) .to ('sqlite::memory:')
+        // REVISIT: how to better configure new sqlite
+        const db = await cds.deploy (_model) .to ({impl: '@cap-js/sqlite'}) // 'sqlite::memory:'
```
Comment (chgeo) [inline test/messaging.test.js:14]: Same

File: test/localized-data/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "devDependencies": {
+    "@cap-js/sqlite": "^1"
```
Comment (chgeo) [inline test/localized-data/package.json:3]: Here, we can configure the test through a normal package.json.
The extra file a bit cumbersome though.

File: test/hierarchical-data.test.js (hunk 1 above)
Comment (johannes-vogel) [reply test/hierarchical-data.test.js:20]: Good question as only the plugin configures itself as the db and that's not run here.

File: test/messaging.test.js (hunk 2 above)
Comment (johannes-vogel) [reply test/messaging.test.js:14]: I think that's different as it uses cds.test on root and we could configure the plugin within the root package.json.

File: test/hierarchical-data.test.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ describe('cap/samples - Hierarchical Data', ()=>{
 	`
 	const model = cds.compile.for.nodejs(csn)
 	const {Categories:Cats} = model.definitions
-	const {expect} = cds.test
+	const { expect } = cds.test.in(__dirname,'..')
```
Comment (johannes-vogel) [inline test/hierarchical-data.test.js:15]: To me that seems like a fair workaround.
I see it as a workaround as we typically require a common structure for cds.test which we do not have in these tests. Hence, we can use the top level package.json as source.

**Comment by johannes-vogel** [comment]
It's fine for me as already stated in comment above but @danjoa wanted to have another look.
I'd propose to merge this one for now and adapt later if required.

**Review by johannes-vogel — APPROVED** [review]

**Pull request merged**